docs/ParameterValidation.md

```diff
@@ -0,0 +1,174 @@
+  - Invalid input: input that does not parse or is out of range
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid rating parameter
+
+- Name: actorId
+- Type: string
+- Parameter Validation
+  - Valid input : well formed actorId rule - starts with 'nm' followed by 5-9 digits, parsed as an int
+    - Example: 'nm1265067'
+    - Status : 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by actorid as [{}, {},..], empty array, when no results, where actorId was in movie
+  - Invalid input: values not conforming to well formed actorid rule.  
+    - Example: 'nm1234', 'ab1234'
+    - Status : 400
+    - Content-Type: text/plain
+    - Response Body: Invalid actorid parameter
+
+- Name: genre
+- Type: string
+- Parameter Validation:
+  - Valid input: genre length [3, 20]
+    - Example: 'War' , 'Documentary'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by genre as [{}, {},..], empty array when no results,  where movie.genres contains genre
+  - Invalid input: genre length < 3 or >20 characters
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid genre parameter
+
+- Name: pageSize
+- Type: integer
+- Description: pageSize limits the number of results returned from the server to a specified number of results
+- Parameter Validation:
+  - Default value: 100
+  - Valid input range: [1, 1000]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies limited to pageSize as [{}, {},..], empty array when no results
+  - Invalid input: input that does not parse
+    - Example: 100.23
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid pageSize parameter
+
+- Name: pageNumber
+- Type: integer
+- Description: pageNumber sets the first position to return from the results of the query
+- Parameter Validation:
+  - Default value: 1
+  - Valid input range: [1, 10000]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies based on pageNumber as [{}, {},..], empty array when no results
+  - Invalid input - input that does not parse
+    - Example: 100.23
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid pageNumber parameter
+
+### API Parameters
+
+- API /api/movies/{movieId}
+- Name: movieId
+- Type: string
+- Parameter Validation:
+  - Valid input: well formed movieId rule - starts with 'tt' followed by 5-9 digits, parsed as an int
+    - Example 'tt0114746'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: Single movie document as JSON
+  - Invalid input: movieId not conforming to well formed movieId rule
+    - Example:'tt1234', 'nm1234'
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid movieId parameter
+  - Invalid input: movieId does not exist
```

Comment:
This isn't really invalid input - it's not found - should it be Valid input: movieId does not exist instead?